### PR TITLE
Spiders on web now ignore an absence of gravity.

### PR DESCRIPTION
**Pull request by Gurkenglas**
## About The Pull Request

Being on a web (not next to one, as is the case with rods in space) lets spiders ignore an absence of gravity.

## Why It's Good For The Game

It's in-flavor and it lets broodmothers that spawn into zero gravity lay a web to move around. Yes, I'm salty.

## Changelog
:cl:
add: Spiders on web now ignore an absence of gravity.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm [hunk 1]
```diff
@@ -101,6 +101,11 @@
 	GLOB.spidermobs -= src
 	return ..()
 
+/mob/living/simple_animal/hostile/poison/giant_spider/mob_negates_gravity()
+	. = ..()
+	var/obj/structure/spider/stickyweb/web = locate() in loc
+	return (. || web)
```
Comment (Cyberboss) [inline code/modules/mob/living/simple_animal/hostile/giant_spider.dm:107]: Avoiding the locate if possible.

```suggestion
	. = ..()
	if (.)
		return
	var/obj/structure/spider/stickyweb/web = locate() in loc
	return web
```

**Review by Cyberboss — CHANGES_REQUESTED** [review]
Nice work

**Comment by bluetshirtguy** [comment]
Spiders are pretty dangerous as is.

**Comment by LemonInTheDark** [comment]
Has god damn soul, finally some good ided prs

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm (hunk 1 above)
Comment (Mothblocks) [reply code/modules/mob/living/simple_animal/hostile/giant_spider.dm:107]:
```suggestion
	return ..() || (locate(/obj/structure/spider/stickyweb) in loc)
```

...works too 😄

**Review by Mothblocks — APPROVED** [review]

**Comment by tgstation-server** [comment]
Your Fix/Feature pull request delta is currently below zero (-5). Maintainers may close future Feature/Tweak/Balance PRs. Fixing issues or helping to improve the codebase will raise this score.

**Comment by ATH1909** [comment]
we could probably let this work for spiders who're standing next to webs as well, tbh

**Pull request merged**